queries/odin/rainbow-delimiters.scm

```diff
@@ -0,0 +1,91 @@
+(parameters
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(tuple_type
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(parenthesized_expression
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(polymorphic_parameters
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(polymorphic_type
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(attribute
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(call_expression
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(index_expression
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
+
+(block
+  "{" @delimiter
+  "}" @delimiter @sentinel) @container
+
+(array_type
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
+
+(struct
+  "{" @delimiter
+  "}" @delimiter @sentinel) @container
+
+(struct
+  "(" @delimiter
+  ")" @delimiter @sentinel) @container
+
+(struct
+  "[" @delimiter
+  "]" @delimiter @sentinel) @container
```

Comment:
Could these three pattern be merged into one? Their delimiters are all on the same level, so they should have the same highlighting. You might have to make some of them optional though.